### PR TITLE
Coordinate-free quasisymmetry objective

### PR DESCRIPTION
**Pull request by landreman**
This PR is for the coordinate-free quasisymmetry objective function, discussed in Simons Hour on August 5. It allows for VMEC-based quasisymmetry optimization without any transformation to Boozer coordinates. It is implemented via a `QuasisymmetryRatioError` class in a new file `vmec_diagnostics.py`. The coverage report for this new file shows 97% test coverage, and the extensive CI tests all pass in my fork. Down the road, the `vmec_diagnostics.py` file could be a home for other similar functions/classes, which compute something in python from a VMEC solution but are not central to running VMEC.

This PR also includes some minor cleanup of other documentation files.

### CODE REVIEW

File: src/simsopt/mhd/vmec_diagnostics.py
Comment (mbkumar) [inline src/simsopt/mhd/vmec_diagnostics.py:108]: Can we use a more descriptive name for normalized toroidal flux argument instead of "s"?

File: src/simsopt/mhd/vmec_diagnostics.py
Comment (mbkumar) [inline src/simsopt/mhd/vmec_diagnostics.py:116]: Line 116 implies that this class needs to be modified for graph framework. I'll take care of it after merging into the master.

File: tests/mhd/test_vmec_diagnostics.py
Comment (mbkumar) [inline tests/mhd/test_vmec_diagnostics.py:67]: Quick look at line 67 tells me that m and n should also be replaced with more descriptive names. However I'd defer to your choice.

File: tests/mhd/test_vmec_diagnostics.py
Comment (mbkumar) [inline tests/mhd/test_vmec_diagnostics.py:80]: With the increasing number of input files in TEST_DIR, I think a README in TEST_DIR containing a short summary of the each  input file may be helpful for future developers or for those interested in knowing more about them.

**Review by mbkumar — COMMENTED** [review]
Hello Matt,

I only have some minor comments. Otherwise everything looks good.

File: src/simsopt/mhd/vmec_diagnostics.py
Comment (landreman) [reply src/simsopt/mhd/vmec_diagnostics.py]: Renamed `s` to `surfaces`.

File: tests/mhd/test_vmec_diagnostics.py
Comment (landreman) [reply tests/mhd/test_vmec_diagnostics.py]: Renamed to `helicity_m` and `helicity_n`.

File: tests/mhd/test_vmec_diagnostics.py
Comment (landreman) [reply tests/mhd/test_vmec_diagnostics.py:84]: I added a README.

**Comment by landreman** [comment]
All the requested changes have been implemented.

**Review by mbkumar — APPROVED** [review]
Thanks Matt!

**Pull request merged**
